tests/nlu/test_components.py

```diff
@@ -220,3 +222,146 @@ async def test_validate_component_keys_raises_warning_on_invalid_key(
+                {"name": "DIETClassifier"},
+            ],
+            True,
+        ),
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "DIETClassifier"},
+            ],
+            False,
+        ),
+    ],
+)
+def test_warn_of_competing_extractors(
+    pipeline_template: List[Dict[Text, Text]], should_warn: bool
+):
+    config = RasaNLUModelConfig({"pipeline": pipeline_template})
+    trainer = Trainer(config)
+
+    if should_warn:
+        with pytest.warns(UserWarning):
+            rasa.nlu.components.warn_of_competing_extractors(trainer.pipeline)
+    else:
+        with pytest.warns(None) as records:
+            rasa.nlu.components.warn_of_competing_extractors(trainer.pipeline)
+
+        assert len(records) == 0
+
+
+@pytest.mark.parametrize(
+    "pipeline_template,data_path,should_warn",
+    [
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "RegexEntityExtractor"},
+                {"name": "DIETClassifier"},
+            ],
+            "data/test/overlapping_regex_entities.yml",
+            True,
+        ),
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "RegexEntityExtractor"},
+            ],
+            "data/test/overlapping_regex_entities.yml",
+            False,
+        ),
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "DIETClassifier"},
+            ],
+            "data/test/overlapping_regex_entities.yml",
+            False,
+        ),
+        (
+            [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "LexicalSyntacticFeaturizer"},
+                {"name": "RegexEntityExtractor"},
+                {"name": "DIETClassifier"},
+            ],
+            "data/examples/rasa/demo-rasa.yml",
+            False,
+        ),
+    ],
+)
+def test_warn_of_competition_with_regex_extractor(
+    pipeline_template: List[Dict[Text, Text]], data_path: Text, should_warn: bool
+):
+    training_data = rasa.shared.nlu.training_data.loading.load_data(data_path)
+
+    config = RasaNLUModelConfig({"pipeline": pipeline_template})
+    trainer = Trainer(config)
+
+    if should_warn:
+        with pytest.warns(UserWarning):
+            rasa.nlu.components.warn_of_competition_with_regex_extractor(
+                trainer.pipeline, training_data
+            )
+    else:
+        with pytest.warns(None) as records:
+            rasa.nlu.components.warn_of_competition_with_regex_extractor(
+                trainer.pipeline, training_data
+            )
+
+        assert len(records) == 0
+
+
+async def test_warn_for_overlapping_entities(tmp_path: Path):
```

Comment:
I'd be in favor of splitting this test following the paradigm of "Unit Tests Should Verify a Single-Use Case" (https://dzone.com/articles/unit-testing-best-practices-how-to-get-the-most-ou) into
* `test_warn_for_overlapping_entities`
* `test_no_duplicate_warning_for_overlapping_entities`

This makes it easier to find the issue when the test fails instead of having to understand the test first.

Reply:
done